### PR TITLE
add check for brc mode when creating the config

### LINKED ISSUE #35: Encoding doesn't check the input rate_control attribute
When the upper-middleware passes the attribute that is not supported by the driver, the driver doesn't check it and still continue the encoding. This is wrong.
For example:  CBR is not exported. The upper-middleware passes the CBR  rate_control. And the driver still helps to do the CBR encoding. 

Expected behaviour:  return error code when the unsupported attribute is passed.

### PR DESCRIPTION
**Pull request by QuPengfei**
Fixes #35

Signed-off-by: Pengfei Qu <Pengfei.Qu@intel.com>

### CODE REVIEW

File: src/i965_drv_video.c [hunk 1]
```diff
@@ -1281,6 +1281,18 @@ i965_CreateConfig(VADriverContextP ctx,
...
+        VAConfigAttrib attrib, *attrib_found;
+        attrib.type = VAConfigAttribRateControl;
+        attrib_found = i965_lookup_config_attribute(obj_config, attrib.type);
+        if ((entrypoint == VAEntrypointEncSlice) &&
+            !(attrib_found->value & i965->codec_info->h264_brc_mode))
+            vaStatus = VA_STATUS_ERROR_INVALID_CONFIG;
+        else if ((entrypoint == VAEntrypointEncSliceLP) &&
+                 !(attrib_found->value & i965->codec_info->lp_h264_brc_mode))
+            vaStatus = VA_STATUS_ERROR_INVALID_CONFIG;
+    }
```
Comment (xhaihao) [inline src/i965_drv_video.c]: It is possible the codec is not h264.

**Comment by xhaihao** [comment]
@uartie, Could you help to check what is the error in CI machine?

File: src/i965_drv_video.c (hunk 1 above)
Comment (QuPengfei) [inline src/i965_drv_video.c]: agree, other codec will have the issue.

**Comment by uartie** [comment]
@xhaihao, it looks like there was a conflict in the container name that we used to compile this on CI.  Not sure how the conflict occurred.  I will go ahead and kick off another build.

**Comment by QuPengfei** [comment]
update the patch

**Comment by xhaihao** [comment]
I think we should do similar check for other vp8/vp9/hevc.

**Comment by QuPengfei** [comment]
@xhaihao  agree to add for vp8/vp9/hevc. i suggest adding support in a separate patch.

**Comment by xhaihao** [comment]
@QuPengfei #35 isn't a AVC specific issue. could you update the commit log if you want to add support for vp8/vp9/hevc in a separate patch?

**Comment by QuPengfei** [comment]
update the commit message

**Pull request merged**